demo/demo_primary.py

```diff
@@ -93,24 +97,26 @@ def clean_slate(
     c_interface=False):
   """
   """
-
+  
   global primary_ecu
-  global client_directory
+  global CLIENT_DIRECTORY
   global _vin
   global _ecu_serial
   global listener_thread
   global use_can_interface
+  global TEMP_DIRECTORY
+  atexit.register(clean_up) # To delete the temp pinned file and folder after
```

Comment:
I'd rather not have this happen in top-level module code, where it will trigger when imported. Better to have this happen immediately before the creation of the directory, which still means you can be sure that it'll be deleted even if an error occurs.

The same goes for demo_secondary.